src/test/java/com/test/combopack/moc/ItemProvider.java

```diff
@@ -6,66 +6,38 @@
 
-	public static List<Box<Item>> availableBoxs=Arrays.asList(
-			new Box<Item>(1, 81d),
-			new Box<Item>(2, 8d),
-			new Box<Item>(3, 75d),
-			new Box<Item>(4, 56d)
-			);
-			
-	
-	public static List<Item> getAvailableItems(int id) {
-		
-		switch (id) {
-		case 1:
-			return 
-					Arrays.asList(
-							new Item(1, 53.38, 45d), 
-							new Item(2, 88.62,98d), 
-							new Item(3, 78.48, 3d), 
-							new Item(4, 72.30, 76d),
-							new Item(5, 30.18, 9d),
-							new Item(6, 46.34, 48d));
-
-		case 2:
-			return 
-					Arrays.asList(new Item(1, 15.3, 34d));
-
-		case 3:
-			return 
-					Arrays.asList(
-							new Item(1, 85.31, 29d), 
-							new Item(2, 14.55,74d), 
-							new Item(3, 3.98, 16d),
-							new Item(4, 26.24, 55d),
-							new Item(5, 63.69, 52d), 
-							new Item(6, 76.25, 75d), 
-							new Item(7, 60.02, 74d), 
-							new Item(8, 93.18, 35d), 
-							new Item(9, 89.95, 78d));
-		case 4:
-			return 
-					Arrays.asList(
-							new Item(1, 90.72, 13d), 
-							new Item(2, 33.80,40d), 
-							new Item(3, 43.15, 10d), 
-							new Item(4, 37.97, 16d),
-							new Item(5, 46.81, 36d), 
-							new Item(6, 48.77, 79d), 
-							new Item(7, 81.80, 45d), 
-							new Item(8, 19.36, 79d), 
-							new Item(9, 6.76, 64d));
-		case 0:
-			return 
-					Arrays.asList(
-							new Item(1, 0.5, 1d), 
-							new Item(2, 0.5, 2d));
-
-		}
-
-		return null;
-	}
+    public static List<Box<Item>> availableBoxs = 
+            Arrays.asList(
+                    new Box<Item>(1, getAvailableItems(1), 81d),
+                    new Box<Item>(2, getAvailableItems(2), 8d), 
+                    new Box<Item>(3, getAvailableItems(3), 75d), 
+                    new Box<Item>(4,getAvailableItems(4), 56d));
+
+    public static List<Item> getAvailableItems(int id) {
+
+        switch (id) {
+            case 1:
+                return Arrays.asList(new Item(1, 53.38, 45d), new Item(2, 88.62, 98d), new Item(3, 78.48, 3d),
+                        new Item(4, 72.30, 76d), new Item(5, 30.18, 9d), new Item(6, 46.34, 48d));
+
+            case 2:
+                return Arrays.asList(new Item(1, 15.3, 34d));
+
+            case 3:
+                return Arrays.asList(new Item(1, 85.31, 29d), new Item(2, 14.55, 74d), new Item(3, 3.98, 16d),
+                        new Item(4, 26.24, 55d), new Item(5, 63.69, 52d), new Item(6, 76.25, 75d), new Item(7, 60.02,
+                                74d), new Item(8, 93.18, 35d), new Item(9, 89.95, 78d));
+            case 4:
+                return Arrays.asList(new Item(1, 90.72, 13d), new Item(2, 33.80, 40d), new Item(3, 43.15, 10d),
+                        new Item(4, 37.97, 16d), new Item(5, 46.81, 36d), new Item(6, 48.77, 79d), new Item(7, 81.80,
+                                45d), new Item(8, 19.36, 79d), new Item(9, 6.76, 64d));
+            case 0:
+                return Arrays.asList(new Item(1, 0.8, 1d), new Item(2, 0.8, 2d));
+
+        }
+
+        return null;
+    }
 }
```

Comment:
**Issue**: The refactor introduces a more dynamic way of initializing available boxes and items, which is a positive change. However, there are a few areas that could be improved for better performance, maintainability, and clarity. Specifically, the method 'getAvailableItems' is now called multiple times for each box initialization, which could be optimized. Additionally, there's a potential issue with scalability and maintainability as the switch case grows. <br> **Fix**: Consider caching the results of 'getAvailableItems' for each ID during the application's lifecycle or request scope, depending on the use case, to avoid redundant computations. Also, think about using a more scalable approach than switch statements, such as a map or dictionary where item lists are keyed by their ID, which can easily be looked up and will simplify adding more items in the future. <br> **Code Suggestion**: 
 ```
 -        switch (id) {
 +        Map<Integer, List<Item>> itemsMap = new HashMap<>();
 +        if (!itemsMap.containsKey(id)) {
 +            // Initialize and put the items list based on ID in the map
 +        }
 +        return itemsMap.get(id);
 ```